contracts/orderbook/Logic/OrderBookData.sol

```diff
@@ -0,0 +1,117 @@
+pragma solidity ^0.8.0;
+import "../Storage/OrderBookEvents.sol";
+import "../Storage/OrderBookStorage.sol";
+
+interface IERC {
+    function approve(address spender, uint amount) external; //for USDT
+}
+
+contract OrderBookData is OrderBookEvents, OrderBookStorage {
+    using EnumerableSet for EnumerableSet.Bytes32Set;
+
+    function initialize(address target) public onlyOwner {
+        _setTarget(this.adjustAllowance.selector, target);
+        _setTarget(this.getActiveOrders.selector, target);
+        _setTarget(this.getActiveOrdersLimited.selector, target);
+        _setTarget(this.getOrderByOrderID.selector, target);
+        _setTarget(this.getActiveOrderIDs.selector, target);
+        _setTarget(this.getTotalOrders.selector, target);
+        _setTarget(this.getActiveOrders.selector, target);
+        _setTarget(this.getTotalActiveOrders.selector, target);
+        _setTarget(this.getOrders.selector, target);
+        _setTarget(this.getOrdersLimited.selector, target);
+    }
+
+    function adjustAllowance(address spender, address token) external pausable {
+        require(
+            protocol.isLoanPool(spender) ||
+                address(protocol) == spender ||
+                vault == spender,
+            "OrderBook: invalid spender"
+        );
+        IERC(token).approve(spender, 0); //needs to be zeroed out because of different iterations of ERC-20 standard
```

Comment:
safeApprove I think does those 2 lines for you

Reply:
yes but safeApprove fails for ERC20s that do not have a return on approve (at least for USDT it does).